snakemake/deployment/conda.py

```diff
@@ -230,18 +230,31 @@ def address(self):
         as default.
 
         """
-        hash = self.hash
-        env_dir = self._env_dir
-        # The underscore prefix cause the conda to skip shortcut installation so
-        # no admin prevleges are needed for windows.
-        get_path = lambda h: os.path.join(env_dir, "_" + h)
-        hash_candidates = [hash[:8], hash]  # [0] is the old fallback hash (shortened)
-        exists = [os.path.exists(get_path(h)) for h in hash_candidates]
-        if self.is_containerized or exists[1] or (not exists[0]):
-            # containerizes, full hash exists or fallback hash does not exist: use full hash
-            return get_path(hash_candidates[1])
-        # use fallback hash
-        return get_path(hash_candidates[0])
+        if self.is_named:
+            return self.name
+        else:
+            hash = self.hash
+            env_dir = self._env_dir
+            get_path = lambda h: os.path.join(env_dir, h)
+            hash_candidates = [
+                hash[:8],
+                hash,
+                hash + "_",  # activate no-shortcuts behavior (so that no admin rights are needed on win)
```

Comment:
@johanneskoester, Maybe I misunderstand the code here, but I think the environment name should be *prefixes* with an underscore "_". I am not sure that postfix does the same thing.